VDF.Core/ScanEngine.cs

```diff
@@ -228,37 +272,27 @@ public sealed class ScanEngine {
 					while (pauseTokenSource.IsPaused) Thread.Sleep(50);
 
 					var entry = ScanList[i];
+					Dictionary<double, byte[]?>? flippedGrayBytes = null;
+					if (Settings.CompareHorizontallyFlipped)
+						flippedGrayBytes = createFlippedGrayBytes(entry);
+
 					for (var n = i + 1; n < ScanList.Count; n++) {
-						var compItem = ScanList[n];
-						if (entry.IsImage && !compItem.IsImage) continue;
-						var duplicateCounter = 0;
-						float[] percent;
-						if (entry.IsImage) {
-							percent = new float[1];
-							percent[0] = GrayBytesUtils.PercentageDifference(entry.grayBytes[0]!, compItem.grayBytes[0]!);
-							if (percent[0] < percentageDifference)
-								duplicateCounter++;
+						FileEntry compItem = ScanList[n];
+						DuplicateFlags flags = DuplicateFlags.None; 
+						bool dupplicate = isDuplicate(entry, null, compItem, differenceLimit, out var difference);
+						if (!dupplicate && Settings.CompareHorizontallyFlipped)
```

Comment:
First condition is false when one item is an image and the other is a video file. In this case you will call "isDuplicate" again which isn't needed.

Reply:
Two questions:
1) https://github.com/0x90d/videoduplicatefinder/blob/364b094355bf2c351ba1de1b3a417ac4f3a329c0/VDF.Core/ScanEngine.cs#L233 I had taken over from you.
But as we just talk about it. Is this really what we want?
If entry is a video it would still be compared with a compItem-Image. 
So shouldn't it be  `if (entry.IsImage != compItem.IsImage) continue; `

_2) Because _isDuplicate()_ contains the complete comparison the image/video check is included there, but right, so the function is called in principle unnecessarily a second time.
So if "unnecessarily" calls should be reduced at all, the image/video check would need to be placed outside the function at the beginning oft the loop. This would also prevent the first "unnecessarily" call. But in consequence the image/video check inside the function then becomes unnecessary - but removing the check from the function would make the function a bit incomplete without the context it is called from....
Probably too many thoughts on this topic, I know....
So if you say nothing else about it I just add the image/video check to line 283 requested. (After (1) is clarified)_